pkg/config/config.go

```diff
@@ -271,6 +274,8 @@ func NewConfig(nn types.NamespacedName, uid types.UID, version, channel string,
 		warnings = append(warnings, saAnnotationWarnings...)
 	}
 
+	spiceConfig.ServiceAccountName = serviceAccountNameKey.pop(config)
```

Comment:
Tiniest nit that most of these simple fields just get set directly where `spiceConfig` is defined.